src/framework/uicomponents/qml/Muse/UiComponents/internal/StyledDropdownView.qml

```diff
@@ -55,6 +55,10 @@ DropdownView {
     property int contentWidth: root.itemWidth
     property int contentHeight: content.contentBodyHeight
 
+    property string typeAheadStr: ""
+    property string lastTypedChar: ""
+    property int lastHighlightedIndex: -1
```

Comment:
I don't think you need `lastHighlightedIndex`. The `currentIndex` is the one that is currently highlighted. If you need to highlight a different one then you simply update the `currentIndex` to point to it.

There is always one `currentIndex` and it must always be the highlighed index.